client.go

```diff
@@ -57,6 +58,8 @@ type Client struct {
 	maximumSize json.Number
 }
 
+var _ Clientable = &Client{}
```

Comment:
I'm not familiar with Go....yet, but this does look like a better / more standard way to test that `Client` implements `Clientable`.
 * http://stackoverflow.com/questions/10498547/ensure-a-type-implements-an-interface-at-compile-time-in-go
Perhaps we could just remove `TestInterfaceIsImplemented` below?